### DISCUSSION TITLE
mixing match-tab and match with send-text

### DISCUSSION THREAD

**Discussion by marconetto** [kovidgoyal/kitty#5834] [Q&A]
I'm wondering if it is possible to mix both match-tab and match options when sending text to multiple windows.

I would like to send texts only to windows that have title ``zsh`` that are in my current focused tab:

Until recently I was using the following command to send text to ``zsh`` windows:
```kitty @ send-text   --match title:zsh  'mytext'```

However, I realized that such command was sending to all windows of all tabs. 

Then I started to use the following command, with match-tab. However, by doing so, it seems that ``--match title:zsh`` was ignored; i.e. all windows (including those without the title ``zsh``) of the current tab were receiving 'mytext'.

```kitty @ send-text  --match-tab recent:0 --match title:zsh  'mytext'```

I tried to invert the match-tab and match to see if there was any priority but the result was the same.

I'm using kitty 0.26.5 on macos.

**Comment by kovidgoyal** [comment]
You can combine boolean expressions in --match so combine your title condition with parent_active see the help for --match in kitty @ send-text --help

**Reply by marconetto** [reply]
Thanks by using the parent_active as state it works perfectly fine:

```kitty @ send-text   --match 'title:zsh and state:parent_active' 'mytext'```
